packages/docs/docs/self-hosting/monitoring.md

```diff
@@ -0,0 +1,128 @@
+
+## Load Balancer Metrics
+
+These metrics measure what users actually experience and provide the most accurate view of external performance.
+
+### Traffic and Performance
+**Request Count** tracks total incoming requests. Compare with server-level metrics to identify discrepancies that might indicate routing issues.
+
+**Response Time** measured at the load balancer level reflects actual user experience better than server-side measurements. Track both median (P50) and 99th percentile (P99) response times.
+
+### Error Tracking
+**HTTP Error Responses** (4xx, 5xx) measured at the load balancer are the definitive source for user-facing errors. Pay special attention to 504 timeouts, which occur when the server takes too long to respond even if the operation eventually succeeds.
+
+**Alert Recommendation**: Set up alerts for 504 errors with a threshold around 3 per minute.
+
+## Database Metrics
+
+PostgreSQL metrics provide detailed insight into data layer performance.
+
+### Resource Usage
+**CPU Utilization** is critical, especially for the writer instance. Since there's only one writer (single point of failure), keep writer CPU below 60-70%. Reader instances can run hotter since you can scale horizontally by adding more readers.
+
+**Connection Count** tracks active database connections. PostgreSQL creates roughly one process per connection, so monitor to avoid resource exhaustion.
+
+### Query Efficiency
+**Tuples Read/Written** measure database activity at the row level. A "tuple" is essentially a row, though it can also exist in indexes.
+
+**Tuples Scanned vs. Fetched** reveals query efficiency. When scanned significantly exceeds fetched, the database is loading data it ultimately discards, indicating inefficient queries or missing indexes.
+
+**Transactions Per Second** shows overall database activity. The metric splits between commits (successful) and rollbacks (failed). Spikes in rollbacks usually indicate problems.
+
+## Cache Metrics (Redis)
+
+Redis serves as a fast in-memory cache to reduce database load.
+
+### Performance
+**CPU and Memory Usage** track Redis resource consumption. Memory usage directly correlates with stored data since Redis is an in-memory store.
+
+**Cache Hit/Miss Ratio** shows cache effectiveness. Some misses are normal, but unusual spikes might indicate application issues or invalid reference patterns.
+
+### Capacity Management
+**Evictions** occur when Redis runs out of memory and must remove least-recently-used items. Ideally this should be zero, but occasional evictions aren't necessarily problematic if your working set still fits in cache.
+
+**Alert Recommendation**: Monitor for evictions > 0 as an early warning of capacity issues.
+
+## Container Metrics
+
+If using containerized deployment (Docker, Kubernetes, ECS, etc.), these metrics track resource usage at the container level.
```

Comment:
These are more accurately "host metrics", since they could be coming from a bare metal system, a VM, or a container